crates/trie/trie/src/node_iter.rs

```diff
@@ -279,6 +303,7 @@ where
                         );
 
                         self.should_check_walker_key = false;
+                        self.last_next_result = None;
```

Comment:
not sure why do we need to reset it? I may be missing something, but I think it's not needed, as nothing can invalidate a cached `next()` result, because we store both the key and the value that it returned.

Reply:
Yeah, makes sense, I removed it